frontend/src/App.vue

```diff
@@ -9,7 +9,7 @@
 </script>
 
 <template>
-  <TheHeader/>
+  <TheHeader :key="$route.fullPath"/>
```

Comment:
Added a key so that the Header rerenders when the route is changed. This is necessary to ensure that the Sign Out button disappears when the user is no longer authenticated.